EIPS/eip-7160.md

```diff
@@ -0,0 +1,163 @@
+  event TokenUriUnpinned(uint256 indexed tokenId, address indexed sender);
+
+  function tokenURIs(uint256 tokenId) external view returns (uint256 index, string[] memory uris);
+  function pinTokenURI(uint256 tokenId, uint256 index) external;
+  function unpinTokenURI(uint256 tokenId) external;
+  function hasPinnedTokenURI(uint256 tokenId) external view returns (bool pinned);
+}
+```
+
+The `tokenURIs` function returns a tuple containing the pinned URI index and a list of all metadata URIs associated with the specified token.
+
+`tokenURIs` is the core method of a multi-metadata token and the additional methods and events are meant to provide a standard interface to dapps to perform pinning and unpinning actions.
+
+The `pinTokenURI` function allows the contract owner to designate a specific metadata URI as the pinned URI for a token.
+
+The `unpinTokenURI` function allows the contract owner to remove the pin on a specific URI index.
+
+The `tokenURI` function defined in the ERC-721 standard must return the pinned URI or the last URI in the list returned by `tokenURIs` when there is not pinned URI. This ensures backwards compatibility with existing contracts and applications that rely on the single metadata URI.
+
+When adding new URIs it is highly encouraged to implement [ERC-4906](./eip-4906.md) and emit a `MetadataUpdate` event.
+
+See the [Implementation](#reference-implementation) section for an example.
+
+## Rationale
+
+The `tokenURIs` function returns both the pinned URI index and the list of all metadata URIs to provide flexibility in accessing the metadata.
+
+The pinned URI can be used as a default or primary URI for the token, while the list of metadata URIs can be used to access individual assets' metadata within the token. Marketplaces could present these as a gallery or media carousels.
+
+Depending on the implementation, the `pinTokenURI` function allows the contract owner or token owner to specify a particular fixed metadata URI index for a token. This enables the selection of a preferred URI by index from the list of available metadata.
+
+When unpinned, it is recommended to return the last URI for the token. However the behavior in the case of unpinned tokens is at the discretion of the implementation and depends on the specific purpose of the token.
+
+## Backwards Compatibility
+
+This extension is designed to be backward compatible with existing [ERC-721](./eip-721.md) contracts. The implementation of the `tokenURI` method must either return the last URI in the list returned by tokenURIs or the pinned URI.
+
+## Reference Implementation
+
+An open-source reference implementation of the `IERC721MultiMetadata` interface can be provided, demonstrating how to extend an existing [ERC-721](./eip-721.md) contract to support multi-metadata functionality. This reference implementation can serve as a guide for developers looking to implement the extension in their own contracts.
+
+```solidity
+// SPDX-License-Identifier: MIT
```

Comment:
Code included inline in the EIP must be `CC0-1.0` licensed. Code in the `assets/...` directory may be MIT.